<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>612</width>
    <height>490</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>612</width>
    <height>490</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>612</width>
    <height>490</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Feature-based image classificator</string>
  </property>
  <property name="styleSheet">
   <string notr="true"/>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QPushButton" name="codebookBtn">
    <property name="geometry">
     <rect>
      <x>450</x>
      <y>240</y>
      <width>150</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string>Create Codebook</string>
    </property>
    <property name="checkable">
     <bool>false</bool>
    </property>
    <property name="autoRepeatDelay">
     <number>3000</number>
    </property>
    <property name="autoRepeatInterval">
     <number>1000</number>
    </property>
   </widget>
   <widget class="QPushButton" name="trainBtn">
    <property name="geometry">
     <rect>
      <x>450</x>
      <y>400</y>
      <width>150</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string>Train SVM</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>200</y>
      <width>110</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string>Output file:</string>
    </property>
   </widget>
   <widget class="Line" name="line">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>280</y>
      <width>590</width>
      <height>20</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>330</y>
      <width>290</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Images histograms creation method:</string>
    </property>
   </widget>
   <widget class="QComboBox" name="histogramMethod">
    <property name="geometry">
     <rect>
      <x>320</x>
      <y>325</y>
      <width>121</width>
      <height>30</height>
     </rect>
    </property>
   </widget>
   <widget class="QComboBox" name="featureDetectionMethod">
    <property name="geometry">
     <rect>
      <x>230</x>
      <y>33</y>
      <width>121</width>
      <height>30</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>38</y>
      <width>210</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Feature-detection method:</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_6">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>591</width>
      <height>17</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">font-weight:bold;</string>
    </property>
    <property name="text">
     <string>1 - Create keypoints codebook</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_7">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>110</y>
      <width>110</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string>Dataset folder:</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_8">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>300</y>
      <width>111</width>
      <height>17</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">font-weight:bold;</string>
    </property>
    <property name="text">
     <string>2 - Train SVM</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_9">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>360</y>
      <width>101</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string>Codebook file:</string>
    </property>
   </widget>
   <widget class="QToolButton" name="inputFolderBtn">
    <property name="geometry">
     <rect>
      <x>570</x>
      <y>110</y>
      <width>30</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string>...</string>
    </property>
   </widget>
   <widget class="QToolButton" name="outputCodebookBtn">
    <property name="geometry">
     <rect>
      <x>570</x>
      <y>200</y>
      <width>30</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string>...</string>
    </property>
   </widget>
   <widget class="QToolButton" name="inputCodebookBtn">
    <property name="geometry">
     <rect>
      <x>570</x>
      <y>360</y>
      <width>30</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string>...</string>
    </property>
   </widget>
   <widget class="QSlider" name="splitSlider">
    <property name="geometry">
     <rect>
      <x>130</x>
      <y>150</y>
      <width>270</width>
      <height>29</height>
     </rect>
    </property>
    <property name="minimum">
     <number>1</number>
    </property>
    <property name="maximum">
     <number>99</number>
    </property>
    <property name="pageStep">
     <number>10</number>
    </property>
    <property name="value">
     <number>80</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
    <property name="tickPosition">
     <enum>QSlider::TicksBelow</enum>
    </property>
   </widget>
   <widget class="QLabel" name="splitLabel">
    <property name="geometry">
     <rect>
      <x>410</x>
      <y>150</y>
      <width>190</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string>Training 80%, testing 20%</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_10">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>150</y>
      <width>101</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string>Dataset split</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>370</x>
      <y>40</y>
      <width>90</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Min Hessian:</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_11">
    <property name="geometry">
     <rect>
      <x>370</x>
      <y>75</y>
      <width>280</width>
      <height>20</height>
     </rect>
    </property>
    <property name="minimumSize">
     <size>
      <width>280</width>
      <height>20</height>
     </size>
    </property>
    <property name="text">
     <string>Codebook clusters:</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="minHessian">
    <property name="geometry">
     <rect>
      <x>510</x>
      <y>35</y>
      <width>90</width>
      <height>30</height>
     </rect>
    </property>
    <property name="minimum">
     <number>100</number>
    </property>
    <property name="maximum">
     <number>10000</number>
    </property>
    <property name="singleStep">
     <number>100</number>
    </property>
    <property name="value">
     <number>800</number>
    </property>
   </widget>
   <widget class="QSpinBox" name="clusterSize">
    <property name="geometry">
     <rect>
      <x>510</x>
      <y>70</y>
      <width>90</width>
      <height>30</height>
     </rect>
    </property>
    <property name="minimum">
     <number>100</number>
    </property>
    <property name="maximum">
     <number>10000</number>
    </property>
    <property name="singleStep">
     <number>100</number>
    </property>
    <property name="value">
     <number>1000</number>
    </property>
   </widget>
   <widget class="QLineEdit" name="inputFolder">
    <property name="geometry">
     <rect>
      <x>130</x>
      <y>110</y>
      <width>430</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string>/home/user/101_ObjectCategories/</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="outputCodebook">
    <property name="geometry">
     <rect>
      <x>130</x>
      <y>200</y>
      <width>430</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string>/home/user/codebook.cbk</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="inputCodebook">
    <property name="geometry">
     <rect>
      <x>130</x>
      <y>360</y>
      <width>430</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string>/home/user/codebook.cbk</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_12">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>300</y>
      <width>431</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>(Uses setup values from codebook metadata)</string>
    </property>
   </widget>
   <widget class="QLabel" name="inputCodebookLabel">
    <property name="enabled">
     <bool>false</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>390</y>
      <width>431</width>
      <height>51</height>
     </rect>
    </property>
    <property name="text">
     <string>Found:</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
    </property>
    <property name="wordWrap">
     <bool>true</bool>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>612</width>
     <height>25</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusBar">
   <property name="sizeGripEnabled">
    <bool>false</bool>
   </property>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <tabstops>
  <tabstop>featureDetectionMethod</tabstop>
  <tabstop>histogramMethod</tabstop>
  <tabstop>minHessian</tabstop>
  <tabstop>clusterSize</tabstop>
  <tabstop>inputFolder</tabstop>
  <tabstop>inputFolderBtn</tabstop>
  <tabstop>splitSlider</tabstop>
  <tabstop>outputCodebook</tabstop>
  <tabstop>outputCodebookBtn</tabstop>
  <tabstop>codebookBtn</tabstop>
  <tabstop>inputCodebook</tabstop>
  <tabstop>inputCodebookBtn</tabstop>
  <tabstop>trainBtn</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
